crates/oxc_parser/src/lexer/keyword_generated.rs

```diff
@@ -0,0 +1,565 @@
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Implements, "implements"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Target, "target"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Boolean, "boolean"),
+    (Ident, ""),
+    (Ident, ""),
+    (Protected, "protected"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Declare, "declare"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Null, "null"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Is, "is"),
+    (Ident, ""),
+    (Package, "package"),
+    (Ident, ""),
+    (Return, "return"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Module, "module"),
+    (Ident, ""),
+    (Class, "class"),
+    (Ident, ""),
+    (From, "from"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Switch, "switch"),
+    (Asserts, "asserts"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Satisfies, "satisfies"),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+    (Ident, ""),
+];
+
+#[inline]
+pub fn table_match_keyword(s: &str) -> Kind {
+    let slice = s.as_bytes();
+    let extract = unsafe { hash::extract_first_and_last_two_bytes(slice) };
+    let hash_code = hash::hash_u32(extract, HASH_TABLE_SEED);
+    let idx = hash_code as usize % HASH_TABLE_SIZE;
+    let (mut kind, key) = KEYWORD_HASH_TABLE[idx];
+    if s.len() != key.len() || s != key {
```

Comment:
This is where having cmov instead of branches can make a significant difference.
Can try splitting
```rust
if s.len() != key.len() {
    kind = Ident
}
if s != key {
    kind = Ident;
}
```

Or use `cmov crate`(https://docs.rs/cmov/latest/cmov/)?